docs/design/access-control.md

```diff
@@ -0,0 +1,93 @@
+# UX for role-based access control
+
+This design document is a proposal for a Dashboard UI that
+allows users to manage Roles, RolesBindings, ClusterRoles, and 
+ClusterRoleBindings in an intuitive way. This document is a 
+formalization of ideas and discussions from #1441.
+
+## Background
+
+Kubernetes now allows cluster admins to use 
+[role-based access control](http://kubernetes.io/docs/admin/authorization/) to secure their clusters. 
+Permissions are managed using the following Kubernetes Resources:
+* Role
+* ClusterRole
+* RoleBinding
+* ClusterRoleBinding
+
+None of these are reflected in the current version of Dashboard UI. There should a simplified, 
+more user-friendly way to manage a cluster's access control by abstracting these raw concepts.
+
+
+## Abstraction / Simplification
+
+The only difference between Roles and ClusterRoles is that Roles apply to a specific namespace 
+and ClusterRoles apply to all namespaces. The same is true for RoleBindings and ClusterRoleBindings. 
+The UI should help users to understand these concepts and user flows should be designed to enforce
+the correct usage. 
+
+NOTE: The design proposed in this document will combine certain concepts:
```

Comment:
Interesting proposal. I think I like it. Simplifies the problem from user's perspective. We could say in some non-prominent place that * means that this is cluster role object, but overall this looks nice.

WDYT? @rf232 @normalnorman